lib/patches/rake-compiler/rake/baseextensiontask.rb

```diff
@@ -0,0 +1,10 @@
+Truffle::Patching.require_original __FILE__
+
+module Rake
+    class BaseExtensionTask < TaskLib
+      def binary(platform = nil)
+        ext = 'su'
+        "#{File.basename(@name)}.#{ext}"
```

Comment:
Did you add the `basename` part?
In https://github.com/rake-compiler/rake-compiler/blob/9f15620e7db145d11ae2fc4ba032367903f625e3/lib/rake/baseextensiontask.rb#L71 I see it without.

Reply:
I must have been looking at an earlier version:
https://github.com/rake-compiler/rake-compiler/blob/1e2e229b36061ff9c65f6fd9af5f4a6616bb5005/lib/rake/baseextensiontask.rb#L71

Reply:
mmh, I am not sure how we can deal with that change safely :/
Maybe instance_method(:binary).source_location, from there get the method code until `end` and add a clause to the `case platform`?
Or maybe by comparing the version if we can get it from somewhere (but more fragile).

Reply:
I think deciding based on versions and failing for those we don't know what patch to apply is the safest.

Reply:
I've updated this to `binary` method to detect the version, please review.